packages/yoga/src/AutoComplete/web/AutoComplete.jsx

```diff
@@ -200,33 +229,51 @@ const AutoComplete = React.forwardRef(
               second.toLowerCase().indexOf(inputValue)
                 ? -1
                 : 1,
-            )
-            .slice(0, 6);
+            );
 
-          const hasSuggestion = isOpen && Boolean(suggestionList.length);
+          if (!!inputValue && isOpen) {
+            setIsSuggestionsOpen(true);
+          }
 
           return (
             <Wrapper
               className={className}
               style={style}
               full={full}
-              isOpen={hasSuggestion}
+              isOpen={isSuggestionsOpen}
               {...getRootProps()}
               ref={ref}
+              onBlur={handleCloseSuggestions}
             >
               <Input
                 {...props}
+                error={error}
                 full={full}
                 onClean={cleanable => {
                   onClean(cleanable);
                   clearSelection();
+                  handleCloseSuggestions();
                 }}
                 {...getInputProps({
                   onFocus: () => (inputValue.length ? openMenu() : null),
                 })}
+                rightIcon={
+                  isSuggestionsOpen ? (
+                    <ChevronUp
+                      onClick={handleCloseSuggestions}
+                      aria-label="Close suggestions"
+                    />
+                  ) : (
+                    <ChevronDown
+                      onClick={handleOpenSuggestions}
+                      aria-label="Open suggestions"
+                    />
+                  )
+                }
+                ref={inputRef}
               />
-              {hasSuggestion && (
-                <List {...getMenuProps()} full={full}>
+              {isSuggestionsOpen && (
+                <List {...getMenuProps()} full={full} error={!!error}>
```

Comment:
what do you think?
```suggestion
                <List {...getMenuProps()} full={full} error={Boolean(error)}>
```

Reply:
Hey, @diegomarcuz!
Do you think that the use of the `!!` makes the code harder to read? I don't see a real gain in making that change, but I can do it if you think it improves the readability of the code.

Reply:
I think it improves readability, but it is a matter of personal choice, feel free to add.